Installing the SDK Page (#355)
LIT-Protocol/docs: docs/build/getting-started/connecting-to-lit.md

# Connecting to Lit

The `LitNodeClient` is used to connect your session to the Lit network. After initializing a `LitNodeClient` instance, you can use the `connect()` method to establish a connection.

When initializing a `LitNodeClient` instance, you must provide a `litNetwork`.

```tsx
import { LitNodeClient } from '@lit-protocol/lit-node-client';
import { LitNetwork } from '@lit-protocol/constants';

const litNodeClient = new LitNodeClient({
  litNetwork: LitNetwork.DatilDev,
  debug: false,
});

await litNodeClient.connect();
```

### `LitNetwork` Constant

The `LitNetwork` constant contains the past and present Lit networks. The constant is imported from the `@lit-protocol/constants` package. The current networks in the constant can be found [here](https://v6-api-doc-lit-js-sdk.vercel.app/enums/constants_src.LitNetwork.html).

### `LitNodeClient` Flags

> You have the option to pass flags to the `LitNodeClient` instance. These flags are used to configure the Lit network connection. You can find a complete list of flags in the [LitNodeClient Config](https://v6-api-doc-lit-js-sdk.vercel.app/interfaces/types_src.LitNodeClientConfig.html). This example will show you the most common flags: `debug` and `storageProvider`.

Reviewer: ```suggestion
You have the option to pass flags to the `LitNodeClient` instance. These flags are used to configure the Lit network connection. You can find a complete list of flags in the [LitNodeClient Config](https://v6-api-doc-lit-js-sdk.vercel.app/interfaces/types_src.LitNodeClientConfig.html). In this guide we will cover the most common flags: `debug` and `storageProvider`.
```

#### `debug`

The `debug` flag is used to enable or disable debug logging. When enabled, debug logs will be written to the console. This flag will only provide debug logs when executing in a Node.js environment. In a browser environment, the `debug` flag will be ignored.

#### `storageProvider`

The `storageProvider` flag is used to configure the storage provider used by the Lit network. When provided, the session keypair will be stored in the provided storage.

If not provided, a new session keypair will be generated each time the `LitNodeClient` is initialized.

> In a browser environment, the `storageProvider` flag will be ignored. The session keypair will be stored in the browser's local storage. To clear the cached session keypair, you can use the `disconnectWeb3()` function.

Reviewer: ```suggestion
In a browser environment, the `storageProvider` flag will be ignored, and the session keypair will be stored in the browser's local storage. To clear the cached session keypair, you can use the `disconnectWeb3` function (imported from the `@lit-protocol/auth-browser` package) like so:

```ts
import { LitNodeClient } from '@lit-protocol/lit-node-client';
import { LitNetwork } from '@lit-protocol/constants';
import { disconnectWeb3 } from "@lit-protocol/auth-browser";

let litNodeClient;
try {
    litNodeClient = new LitNodeClient({
        litNetwork: LitNetwork.DatilDev,
        debug: false,
    });

    await litNodeClient.connect();
} catch (error) {
  // handle errors
} finally {
    disconnectWeb3();
    litNodeClient.disconnect();
}
```

#### `storageProvider`

The `storageProvider` flag is used to configure the storage provider used by the Lit network. When provided, the session keypair will be stored in the provided storage.

If not provided, a new session keypair will be generated each time the `LitNodeClient` is initialized.

In a browser environment, the `storageProvider` flag will be ignored, and the session keypair will be stored in the browser's local storage. To clear the cached session keypair, you can use the `disconnectWeb3` function (imported from the `@lit-protocol/auth-browser` package) like so:

```ts
import { LitNodeClient } from '@lit-protocol/lit-node-client';
import { LitNetwork } from '@lit-protocol/constants';
import { disconnectWeb3 } from "@lit-protocol/auth-browser";

let litNodeClient;
try {
    litNodeClient = new LitNodeClient({
        litNetwork: LitNetwork.DatilDev,
        debug: false,
    });

    await litNodeClient.connect();
} catch (error) {
  // handle errors
} finally {
    disconnectWeb3();
    litNodeClient.disconnect();

> }

Reviewer: ```suggestion
}
```
```

> ```ts
> import { LitNodeClient } from '@lit-protocol/lit-node-client';
> import { LitNetwork } from '@lit-protocol/constants';
> import { disconnectWeb3 } from "@lit-protocol/auth-browser";
> 
> let litNodeClient;
> try {
>     litNodeClient = new LitNodeClient({
>         litNetwork: LitNetwork.DatilDev,
>         debug: false,
>     });
> 
>     await litNodeClient.connect();
> } catch (error) {
>   // handle errors
> } finally {
>     disconnectWeb3();
>     litNodeClient.disconnect();
> }

Reviewer: ```suggestion
```ts
import { LitNodeClient } from '@lit-protocol/lit-node-client';
import { LitNetwork } from '@lit-protocol/constants';
import { disconnectWeb3 } from "@lit-protocol/auth-browser";

let litNodeClient;
try {
    litNodeClient = new LitNodeClient({
        litNetwork: LitNetwork.DatilDev,
        debug: false,
    });

    await litNodeClient.connect();
} catch (error) {
  // handle errors
} finally {
    disconnectWeb3();
    litNodeClient.disconnect();
}
```

```

> ```ts
> import { LitNodeClient } from '@lit-protocol/lit-node-client';
> import { LitNetwork } from '@lit-protocol/constants';
> import { disconnectWeb3 } from "@lit-protocol/auth-browser";
> 
> let litNodeClient;
> try {
>     litNodeClient = new LitNodeClient({
>         litNetwork: LitNetwork.DatilDev,
>         debug: false,
>     });
> 
>     await litNodeClient.connect();
> } catch (error) {
>   // handle errors
> } finally {
>     disconnectWeb3();
>     litNodeClient.disconnect();
> }

Reviewer: ```suggestion
```ts
import { LitNodeClient } from '@lit-protocol/lit-node-client';
import { LitNetwork } from '@lit-protocol/constants';
import { disconnectWeb3 } from "@lit-protocol/auth-browser";

let litNodeClient;
try {
  litNodeClient = new LitNodeClient({
      litNetwork: LitNetwork.DatilDev,
      debug: false,
  });

  await litNodeClient.connect();
} catch (error) {
  // handle errors
} finally {
  disconnectWeb3();
  litNodeClient.disconnect();
}
```
```
